### ISSUE TITLE
installation on windows does not start

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#85]
_From [nahu...@gmail.com](https://code.google.com/u/111568301475995177306/) on 2009-10-19T10:42:19Z_

What steps will reproduce the problem?

 1. Download spyder 1.0 for window, go through installation. 
2. Go to start menu - look for spyder in the applications. Click to launch. 
3. Nothing happens.

What is the expected output? Spyder should launch. 
Incase the installer finds there's no qt4 installed on the windows machine,
it should terminate (alternatevily, download and install qt4). What version of the product are you using? On what operating system? 1.0.0 on windows. 

Please provide any additional information below

.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=85_

**Comment by spyder-issues-migrator** [comment]
_From [hologra...@gmail.com](https://code.google.com/u/106810255920517492489/) on 2009-10-19T12:22:13Z_

Same thing happening for me. Order of events: 

1. Install Python 2.6 
2. Install Spyder IDE for python version 2.6
3. Attempt to launch Spyder IDE (nothing happens)

Attempted machine restart as well -> Same problem

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2009-10-19T12:31:22Z_

So, I understand that Spyder does not work because you didn't install the required
dependencies on your machine.

I'm sorry but the Python(x,y) plugin installers (Windows) do not support package
manager features like detecting dependencies and eventually installing them, like
every other GUI-based installers on Windows (distutils: .exe and .msi installers).

In other words, if you're insterested in such features, I suggest migrating to an OS
having a package manager system like Debian (Ubuntu): if you choose to install the
Spyder debian package (soon available), it will automatically install all required
and eventually optional dependencies.

On Windows, there are two options:
    - install manually the required dependencies
    - or install a distribution like Python(x,y), which BTW includes Spyder

**Status:** Done

**Comment by spyder-issues-migrator** [comment]
_From [nahu...@gmail.com](https://code.google.com/u/111568301475995177306/) on 2009-10-19T12:47:36Z_

Pier, 
I already use debian. But consider this case:
A windows user sees the software on my laptop. 
He thinks: "Oh, cool! Looks like a nice thing to try". He goes back to his desktop. 
Has no clue what is qt4 or scintilla. So he downloads spyder, clicks next-next-next.
And then tries to lunch spyder. Nothing happens. 
User windows thinks: "Ah, free software, bunch of &%^#@ amatures. I stay with my matlab".

I am not forcing you to make package dependencies solvers to windows. But I've seen
python installers that do check dependencies and download and even install them for
"Bill the Windows User". 
See for example deluge-torrent. It has a nice installer, that installs python-gtk and
gtk, if it can't find them on the system.

I will try sending a patch next month - too busy this one. But maybe you can
implement it faster than me and release a 1.0.1 fix ?

This should be something like:
try:
 import qt
catch:
 "some window: Dear Bill - I could not find a QT distribution on your windows"
 "Would you like to get one (It's essential for spyder to work)?"
 "DOWNLOAD" -> Action: wget some qt distribution, launch installer, next next...   
back to spyder installer. 

I think it's simple to implement. 
 

And I've seen this case happen today, when I came really excited this morning to
work, after reading your scipy-users announce. 
Came to my boss - who is a long time Matlab user, who really tries to like python. I
thought, hmm... that's a cool project for him. Now guess what he thinks. Guess how
stupid I look...

I agree with you that python(x,y) is a good solution. Only it's a big installer and
requires lot's of time to install. Spyder is quick sageway into python for non-linux
users. 

Your work is very well appreciated. I feel spyder has lot's of potential. I like it,
but so far can't use it or recommend it to windows users.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2009-10-19T13:03:09Z_

I see. Thanks for these explanations, I understand now your point of view.

I think I could make an installer including all dependencies even Python itself. It
would be lighter than Python(x,y) full installer, but it would certainly be around
90MB I think (with Python, PyQt4, NumPy, SciPy, matplotlib and PIL for example). It
reminds me of the Python(x,y) Light or Basic editions. Through the Python(x,y) web
installer, it's possible to download these lighter installers.

I'll think about it. But Spyder is a Python module, nothing more, nothing less --
meaning that it requires dependencies.

**Status:** Accepted

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2009-10-20T13:10:11Z_

The more I think about it the less I want to do a specific installer.
Python(x,y) is already out there to solve this kind of issue:
  - "Python(x,y) is a big installer": yes it is, as Matlab is, and modern storage
media aren't afraid of ~400MB :)
  - "Python(x,y) requires a lot of time to install": no, absolutely not! -> every
single package is optional -- if you unselect Eclipse, it's very fast, and you may
unselect more packages if you want to speed-up installation a bit more
So basically, I don't see why you didn't install Python(x,y) on your boss computer ;)
It's the effective way of installing Python for scientific purpose on a Windows machine.

**Status:** Done
